docs/doxygen/overviews/high_dpi.md

```diff
@@ -69,55 +69,194 @@ pixels**, abbreviated as "DIP", that are always of the same size on all
+support (currently only GTK 3 and macOS), logical pixels are the same as DIP,
+but different from physical pixels.
+
+However under all platforms the following functions can be used to convert
+between different kinds of pixels:
+
+* From DIP to logical pixels: use wxWindow::FromDIP() or wxWindow::ToDIP().
+* Logical pixels and physical pixels: multiply of divide by wxWindow::GetContentScaleFactor().
+* From DIP to physical pixels: multiply of divide by wxWindow::GetDPIScaleFactor().
 
+Or, in the diagram form:
 
-High-Resolution Images and Artwork
+@dot
+digraph Pixels
+{
+    node [shape = hexagon, style = filled];
+
+    LP  [fillcolor = lightblue, label = "Logical\nPixels"];
+    DIP [fillcolor = yellow, label = "DI\nPixels"];
+    PP  [fillcolor = green, label = "Physical\nPixels"];
+
+    LP  -> PP [dir = both, minlen = 3, headlabel = "mul by GetContentScaleFactor()", taillabel = "div by GetContentScaleFactor()"];
+    LP -> DIP [dir = both, weight = 2, minlen = 3, headlabel = "ToDIP()", taillabel = "FromDIP()"];
+    DIP -> PP [dir = both, minlen = 15, headlabel = "mul by GetDPIScaleFactor()", taillabel = "div by GetDPIScaleFactor()", constraint = false] ;
+}
+@enddot
+
+
+High-Resolution Images and Artwork  {#high_dpi_artwork}
 ==================================
 
-In order to benefit from the increased detail on High DPI devices you might want
+In order to really benefit from the increased detail on High DPI devices you need
 to provide the images or artwork your application uses in higher resolutions as
 well. Note that it is not recommended to just provide a high-resolution version
 and let the system scale that down on 1x displays. Apart from performance
-consideration also the quality might suffer, contours become more blurry.
-
-You can use vector based graphics like SVG or you can add the same image at different
-sizes / resolutions.
-
-[comment]: # (TODO: API and Use Cases)
-
-Platform-Specific Build Issues
+consideration also the quality might suffer, contours become more blurry, so
+for best results it is recommended to use the images that can be used without
+scaling at the common DPI values, i.e. at least 100% and 200% scaling. If you
+don't want providing several copies of all bitmaps, you can use a single
+vector image in [SVG format][1] instead.
+
+[1]: https://en.wikipedia.org/wiki/Scalable_Vector_Graphics
+
+In either case, you must use wxBitmapBundle class representing several
+different versions of the same bitmap (or even potentially just a single one,
+which makes it upwards-compatible with wxBitmap). Most functions accepting
+wxBitmap or wxImage in wxWidgets API have been updated to work with
+wxBitmapBundle instead, which allows the library to select the appropriate
+size depending on the current DPI and, for the platforms supporting it
+(currently only MSW), even update the bitmap automatically if the DPI changes,
+as can happen, for example, when the window showing the bitmap is moved to
```

Comment:
actually since we transform a bundle into a NSImage with several representations on macOS, the same happens on macOS automatically as well

Reply:
Thanks, changed to "(currently only MSW and macOS)".